src/app/zap-templates/partials/im_command_handler_cluster_commands.zapt

```diff
@@ -8,40 +9,77 @@ const uint8_t * {{asSymbol label}};
 {{else}}
 {{asUnderlyingZclType type}} {{asSymbol label}};
 {{/if}}
+bool {{asSymbol label}}Exists = false;
 {{/zcl_command_arguments}}
+uint32_t validArgumentCount = 0;
 
 while ((TLVError = dataTlv.Next()) == CHIP_NO_ERROR)
 {
   switch (TLV::TagNumFromTag(dataTlv.GetTag()))
   {
 {{#zcl_command_arguments}}
     case {{index}}:
+      if ({{asSymbol label}}Exists)
+      {
+            ChipLogProgress(Zcl, "Ignoring duplicate TLV tag %" PRIx32, TLV::TagNumFromTag(dataTlv.GetTag()));
```

Comment:
A duplicate TLV tag in a struct, which is what we're looking at here (I think; see https://github.com/project-chip/connectedhomeip/issues/5577 for why I can't tell for sure) is invalid.  Shouldn't this just cause us to treat this whole thing as invalid instead of pretending it wasn't there?  @mrjerryjohns

Reply:
Yeah, you cannot have the same tag appear twice in a struct context, that should be flagged as an error.

That is permissible however in a list TLV type.

Reply:
Done, error CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT is used now.